strapi/strapi
# enhancement: create dedicated preview page

### What does it do?

Sets up the admin routes for Preview. The route the edit view URL + `/preview`. So I two routes are technically created, one for single types and one for collection types.

Nothing is currently displayed on the page, I'm breaking it into several PRs.

I set up the page very similarly to what we did in history as it worked pretty well. The page component does all the data fetching that the components will need. This way we can have centralized loading and error states. Then the data is made available to components via a context hook.

Most e2e tests are deleted for now, we'll restore/update them as we introduce the bits of UI in the next PRs.

### Why is it needed?

It's what we need to then display the header and iframe.

### How to test it?

- open the edit view for a saved content type
- the "copy preview url" button should not appear
- click on open preview, you should be redirect to the `/preview` route and see "preview will go here"
- the query params should be kept on the preview route. Try that by clicking on preview from different locales and from both draft and published tabs
